docs: clarify how `esmodules: true` works with `browserslist` config (#2525)
babel/website: docs/preset-env.md

`boolean`.

> You may also target browsers supporting ES Modules (https://www.ecma-international.org/ecma-262/6.0/#sec-modules). When the `esmodules` target is specified, the `browsers` target will be ignored. It _will_, however, intersect with `browserslist`'s targets. You can use this approach in combination with `<script type="module"></script>` to conditionally serve smaller scripts to users (https://jakearchibald.com/2017/es-modules-in-browsers/#nomodule-for-backwards-compatibility).

Reviewer: When I told you that `esmodules` is intersected I was only considering the top-level option :sweat_smile: 

When using `@babel/preset-env`'s option, both `browsers` and `.browserslistrc` are ignored for backward compat reasons.

Author: @nicolo-ribaudo Changed.

You may also target browsers supporting ES Modules (https://www.ecma-international.org/ecma-262/6.0/#sec-modules). When specifying this option, the browsers field will be ignored. You can use this approach in combination with `<script type="module"></script>` to conditionally serve smaller scripts to users (https://jakearchibald.com/2017/es-modules-in-browsers/#nomodule-for-backwards-compatibility).

> > _Please note_: when specifying the esmodules target, browsers targets will be ignored. But `browserslist`'s configuration will still work intersectly at the same time.

Reviewer 2: Howabout:

```suggestion
> _Please note_: when the `esmodules` target is specified, the `browsers` target will be ignored. It _will_, however, intersect with `browserslist`'s targets.
```

Should we remove the similar statement in the paragraph above?

Reviewer: Actually, there is a difference between the top-level `targets` option and the `targets` option of `@babe/preset-env` (we'll align them by changing the `@babel/preset-env` behavior in Babel 8).

- In the `@babel/preset-env` options, `esmodules: true` always overrides anything else: both the `browsers` options and the `.browserslistrc` file
- In the top-level `targets` option (https://babeljs.io/docs/en/options#targets), it will intersect the `browsers` option or `.browserlistrc` file.

However, I just noticed that https://babeljs.io/docs/en/options#targetsesmodules is wrong because it says that `browsers` will be ignored, while it's intersected.

Author: @nicolo-ribaudo I've changed `options.md` at the same time, please review again.

> > _Please note_: when specifying the esmodules target, browsers targets will be ignored. But `browserslist`'s configuration will still work intersectly at the same time.

Reviewer 3: Yeah I think we can merge these two:

```
You may also target browsers supporting ES Modules (https://www.ecma-international.org/ecma-262/6.0/#sec-modules). When the `esmodules` target is specified, the `browsers` target will be ignored. It _will_, however, intersect with `browserslist`'s targets. You can use this approach in combination with `<script type="module"></script>` to conditionally serve smaller scripts to users (https://jakearchibald.com/2017/es-modules-in-browsers/#nomodule-for-backwards-compatibility).
```
